### PR TITLE
FT (Users add ratings): A user must add ratings against an accomodation

### PR DESCRIPTION
**Pull request by Cheza-Dzabala**
#### What does this PR do?

Allows users to provide ratings on accommodations. 

#### Description of Task to be completed?

- Allow a user to provide feedback & ratings on accommodations.

#### How should this be manually tested?

1. Clone this Repository

2. Run ` npm install `

3. Run ` sequelize db:migrate `

4. Run ` sequelize db:seed:all `

5. Run ` npm run dev `

6. Log in with seeded user ` alain@caretbn.com `

7. Attempt to send  ` POST ` request to ` http://Localhost:<PORT>/api/v1/ratings/ `

```javascript 
{
	"accommodationId": INTEGER
	"rating": FLOAT
	"feedback": STRING || NULLABLE
}
```

8. When the booking is successful, you can view all ratings by sending a ` GET ` request to ` http://Localhost:<PORT>/api/v1/ratings/ <:slug>`

#### Any background context you want to provide?
N / A

#### Screenshots (if appropriate)

Screen Shot 2019-10-30 at 12 57 19

Screen Shot 2019-10-30 at 12 52 36

Screen Shot 2019-10-30 at 12 54 26

#### Questions:

N / A

### CODE REVIEW

File: src/middlewares/ratingsMiddleware.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            accommodationId
+        }
+    });
+
+    if (rating.length > 0) {
+        return utilities.responseHelper(
+            res,
+            'You have already rated this facility.',
+            null,
+            403
```
Comment (sabin18) [inline src/middlewares/ratingsMiddleware.js]: the status code is 409.

**Review by sabin18 — CHANGES_REQUESTED** [review]

**Review by nignanthomas — APPROVED** [review]
@Cheza-Dzabala Great job
The checks and validations 👍

**Review by sabin18 — APPROVED** [review]
good job @Cheza-Dzabala

File: src/controllers/ratingsController.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import { storeRatingService } from '../services/ratingsServices';
+import accomodationServices from '../services/accommodationServices';
+import utilities from '../utils/index';
+
+export default class FeebackController {
+
+    static async viewRating(req, res) {
+
+        const { slug } = req.params;
+        const accomodation = await accomodationServices.findOneAccommodation({
```
Comment (nakiwuge) [inline src/controllers/ratingsController.js:14]: Make this a helper function

File: src/routes/api/ratings.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import validateToken from '../../middlewares/auth/validateToken';
+import {
+  checkIfBooked,
+  checkDateOfRating,
+  validateBody,
+  checkIfRated
+} from '../../middlewares/ratingsMiddleware';
+
+const router = new Router();
+
```
Comment (nakiwuge) [inline src/routes/api/ratings.js]: remove this line

File: src/services/bookingServices.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import models from '../database/models/index';
+
```
Comment (nakiwuge) [inline src/services/bookingServices.js:2]: Remove this line

**Review by nakiwuge — CHANGES_REQUESTED** [review]
@Cheza-Dzabala good job. Kindly work on the feedback below.
- Users should be logged in tho view there ratings
- Users should provide feedback to when they give a rating of below 4
- Let us use whole nimbers for rating
- Users should view the average rating when viewing all accommodation and single accommodation.
- The feedback validation of at least 30 characters is too much for a user.
- Add a virtual boolean field `hasRated` when retrieving all accommodation and a single accommodation
Also, check the comments on the pr

**Review by nakiwuge — APPROVED** [review]

**Pull request merged**